apps/client/src/modules/platform/pipelineruns/pages/tekton-result-details/components/Details/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { Terminal } from "@/core/components/Terminal";
+import { TerminalRef } from "@/core/components/Terminal/types";
+import { Card } from "@/core/components/ui/card";
+import { LoadingSpinner } from "@/core/components/ui/LoadingSpinner";
+import { useEffect, useRef } from "react";
+import { useTektonResultPipelineRunLogsQuery } from "../../hooks/data";
+import { routeTektonResultPipelineRunDetails } from "../../route";
+
+export const Details = () => {
+  const params = routeTektonResultPipelineRunDetails.useParams();
+  const logsQuery = useTektonResultPipelineRunLogsQuery();
+  const terminalRef = useRef<TerminalRef>(null);
+  const hasScrolledRef = useRef(false);
+
+  const hasLogs = !!logsQuery.data?.logs;
+  const isError = logsQuery.isError;
+  // Only show loading when actively fetching AND no cached data
+  const showLoading = logsQuery.isFetching && !hasLogs;
+
+  // Scroll to top only on initial load (not on tab switches)
+  useEffect(() => {
+    if (hasLogs && terminalRef.current && !hasScrolledRef.current) {
+      hasScrolledRef.current = true;
+      const timeoutId = setTimeout(() => {
+        const terminal = terminalRef.current?.getTerminal();
+        terminal?.scrollToTop();
+      }, 50);
+      return () => clearTimeout(timeoutId);
+    }
+  }, [hasLogs]);
+
+  return (
+    <Card className="relative overflow-hidden">
+      {/* Centered overlay on terminal area */}
+      {(showLoading || isError || !hasLogs) && (
+        <div className="absolute inset-0 z-10 flex items-center justify-center bg-white/80">
+          <div className="flex flex-col items-center gap-3">
+            {showLoading && (
+              <>
+                <LoadingSpinner size={32} />
+                <span className="text-muted-foreground text-sm">Loading logs...</span>
+              </>
+            )}
+
+            {isError && !showLoading && (
+              <div className="flex flex-col items-center gap-2 text-center">
+                <span className="text-destructive text-sm font-medium">Failed to load logs</span>
+                <span className="text-muted-foreground text-xs">
+                  {logsQuery.error instanceof Error ? logsQuery.error.message : "Unknown error"}
+                </span>
+              </div>
+            )}
+
+            {!showLoading && !isError && !hasLogs && (
+              <span className="text-muted-foreground text-sm">No logs available</span>
+            )}
+          </div>
+        </div>
+      )}
+
+      {/* Terminal always rendered */}
+      <Terminal
+        ref={terminalRef}
+        content={logsQuery.data?.logs || ""}
+        height={600}
+        enableSearch={true}
+        enableDownload={true}
+        enableCopy={true}
+        showToolbar={true}
+        readonly={true}
+        downloadFilename={`pipelinerun-${params.recordUid.slice(0, 8)}-logs.log`}
```

Comment:
move into seperate constant

Reply:
We use it only here. I propose to leave it as it is